v2/concordium/types.proto

```diff
@@ -3255,4 +3255,291 @@ message WinningBaker {
+    optional Address invoker = 1;
+    // Address of the contract instance to invoke.
+    ContractAddress instance = 2;
+    // Amount to invoke the smart contract instance with.
+    Amount amount = 3;
+    // The entrypoint of the smart contract instance to invoke.
+    ReceiveName entrypoint = 4;
+    // The parameter bytes to include in the invocation of the entrypoint.
+    Parameter parameter = 5;
+    // The maximum energy to allow for the invocation. Note that the node imposes an energy
+    // quota that is enforced in addition to this limit.
+    Energy energy = 6;
+}
+
+// An operation that can update the state as part of a dry run.
+message DryRunStateOperation {
+    oneof operation {
+        // Sets the current block time to the given timestamp for the purposes of future
+        // transactions.
+        //
+        // The energy cost of this operation is 50.
+        Timestamp set_timestamp = 1;
+        // Add a specified amount of newly-minted CCDs to a specified account.
+        // The amount cannot cause the total circulating supply to overflow.
+        //
+        // The energy cost of this operation is 400.
+        DryRunMintToAccount mint_to_account = 2;
+        // Dry run a transaction, updating the state if it succeeds.
+        //
+        // The energy cost of this operation is 400 plus the energy used by executing the
+        // transaction.
+        DryRunTransaction run_transaction = 3;
+    }
+}
+
+// Mint a specified amount and credit it to the specified account as part of a dry run.
+message DryRunMintToAccount {
+    // The account to mint to.
+    AccountAddress account = 1;
+    // The amount to mint and credit to the account.
+    Amount amount = 2;
+}
+
+// Dry run an account transaction
+message DryRunTransaction {
+    // The account to use as the sender of the transaction.
+    AccountAddress sender = 1;
+    // The energy limit set for executing the transaction.
+    Energy energy_amount = 2;
+    // The payload of the transaction.
+    AccountTransactionPayload payload = 3;
+    // Which credentials and keys should be treated as having signed the transaction.
+    // If none is given, then the transaction is treated as having one signature for credential 0,
+    // key 0. Therefore, this is only required when the transaction is from a multi-signature
+    // account. There two reason why you might want to specify signatures:
+    //   * The cost of the transaction depends on the number of signatures, so if you want to
+    //     get the correct cost for a multi-signature transaction, then specifying the signatures
+    //     supports this.
+    //   * When changing account keys on a multi-credential account, the transaction must be signed
+    //     by the credential whose keys are being changed.
+    //
+    // Note that the signature thresholds are not checked as part of the dry run. Duplicated
+    // signatures are only counted once.
+    repeated DryRunSignature signatures = 4;
```

Comment:
We should use a map here instead like we do for account signatures (modulo the value type of the map of course).